polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -55,6 +56,18 @@ mlir::Value ValueCategory::getValue(mlir::OpBuilder &builder) const {
 void ValueCategory::store(mlir::OpBuilder &builder, mlir::Value toStore) const {
   assert(isReference && "must be a reference");
   assert(val && "expect not-null");
+  if (toStore.getType().isInteger(1)) {
+    // Ad-hoc extension of booleans
+    auto ElementType =
+        llvm::TypeSwitch<Type, Type>(val.getType())
+            .Case<LLVM::LLVMPointerType>(
+                [](auto Ty) -> Type { return Ty.getElementType(); })
+            .Case<MemRefType>(
+                [](auto Ty) -> Type { return Ty.getElementType(); })
```

Comment:
Does TypeSwitch support fallthrough?

Reply:
It does not. It is just syntactic sugar for a series of `if (auto foo = dyn_cast...)`.